### PR TITLE
PERF make commits faster and use less memory

### PR DESCRIPTION
**Pull request by beckermr**
Trying to get things moving again.

### CODE REVIEW

File: conda_forge_tick/cli.py [hunk 1]
```diff
@@ -43,98 +43,120 @@ def deploy(args):
...
+        # modified and staged but not deleted
+        + subprocess.run(
+            "git diff --name-only --cached --diff-filter=d",
+            shell=True,
+            capture_output=True,
+        ).stdout.decode("utf-8").splitlines()
+    )
+    n_added = 0
+    for file in files_to_add:
+        if file is not None and len(file) > 0 and os.path.exists(file):
```
Comment (CJ-Wright) [inline conda_forge_tick/cli.py:75]: would `bool(file)` suffice to replace the first two clauses

File: conda_forge_tick/cli.py [hunk 2]
```diff
@@ -43,98 +43,120 @@ def deploy(args):
...
-
-    status = 1
-    num_try = 0
-    while status != 0 and num_try < 10 and graph_ok:
+        # make sure the graph can load, if not we will error
         try:
-            if num_try == 9:
+            gx = load_graph()
+            # TODO: be more selective about which json to check
+            for node, attrs in gx.nodes.items():
```
Comment (CJ-Wright) [inline conda_forge_tick/cli.py:93]: Do you want to use your file list here to check only the nodes that have been updated? (If any)

**Review by CJ-Wright — COMMENTED** [review]
some light suggestions

File: conda_forge_tick/cli.py (hunk 1 above)
Comment (beckermr) [reply conda_forge_tick/cli.py:75]: It might but I'd be a lot less clear with a marginal performance gain.

File: conda_forge_tick/cli.py (hunk 2 above)
Comment (beckermr) [reply conda_forge_tick/cli.py:93]: Not worth the effort.

File: conda_forge_tick/cli.py (hunk 1 above)
Comment (beckermr) [reply conda_forge_tick/cli.py]: I simplified a bit here.

**Review by CJ-Wright — APPROVED** [review]

**Pull request merged**